document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -427,7 +432,7 @@ public Optional<Document> update(
       throw new IOException("At least one update is required");
     }
 
-    try (final Connection connection = client.getNewConnection()) {
+    try (final Connection connection = connectionPool.getConnection()) {
```

Comment:
Can we expose getPooledConnection() in PostgresClient and use it here? It abstract out the logic of connection management to PostgresClient, and PostgresCollection doesn't have to deal with it?

Reply:
Makes sense. Will update it.